build-system/tasks/bundle-size/index.js

```diff
@@ -53,23 +57,36 @@ const bundleSizeAppBaseUrl = 'https://amp-bundle-size-bot.appspot.com/v0/';
  * @return {Map<string, number>} the bundle size in KB rounded to 2 decimal
  *   points.
  */
-function getBrotliBundleSizes() {
-  // Brotli compressed size fluctuates because of changes in the RTV number, so
-  // normalize this across pull requests by replacing that RTV with a constant.
+async function getBrotliBundleSizes() {
   const bundleSizes = {};
+  const replacementExpression = new RegExp(internalRuntimeVersion, 'g');
 
   log(cyan('brotli'), 'bundle sizes are:');
-  for (const filePath of globby.sync(fileGlobs)) {
-    const normalizedFileContents = fs
-      .readFileSync(filePath, 'utf8')
-      .replace(new RegExp(internalRuntimeVersion, 'g'), normalizedRtvNumber);
-
-    const relativeFilePath = path.relative('.', filePath);
-    const bundleSize = parseFloat(
-      (brotliSize.sync(normalizedFileContents) / 1024).toFixed(2)
-    );
-    log(' ', cyan(relativeFilePath) + ':', green(`${bundleSize}KB`));
-    bundleSizes[relativeFilePath] = bundleSize;
+  const values = report(process.cwd(), content =>
+    // Brotli compressed size fluctuates because of changes in the RTV number, so
+    // normalize this across pull requests by replacing that RTV with a constant.
+    content.replace(replacementExpression, normalizedRtvNumber)
+  );
+  let next = await values.next();
+  if (next.done) {
+    return bundleSizes;
+  }
+  const fileNames = next.value[0];
+  const paths = new Set(Array.from(fileNames.keys()));
+  while (!next.done) {
+    const {value: currentValue} = next;
+    paths.forEach(filePath => {
+      const brotliSize =
+        currentValue[0].get(filePath) && currentValue[0].get(filePath)[0][0];
```

Comment:
Once Optional Chaining is supported by latest Node LTS, we can switch this over.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
I can't wait for optional chaining :heart_eyes: